opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -175,7 +176,7 @@ namespace Opm {
         /// \param[in, out] reservoir_state   reservoir state variables
         /// \param[in, out] well_state        well state variables
         template <class NonlinearSolverType>
-        IterationReport nonlinearIteration(const int iteration,
+        SimulatorReport nonlinearIteration(const int iteration,
```

Comment:
no, but that's also not the problem: the problem is accessing the report object after an exception is thrown because nothing can returned to the caller.